gems/smithy-schema/lib/smithy-schema/shapes.rb

```diff
@@ -30,28 +30,52 @@ def []=(key, value)
         end
       end
 
+      # A reference to a shape.
+      class ShapeRef
```

Comment:
I'm not sure if this is a good class name for this purpose. I always felt that it was bit off when working with V3. 
While it does contain a reference to the shape - it also contains all the details that are tied to the parent shape. I don't have any suggestion on what is a better name but something I was thinking.

Reply:
Talked offline. I can't really think of anything else but I do think that ShapeRef is actually very descriptive - it contains a reference to the shape as you said. It could be "target" or a method called "target" - I tried that and it didn't sit right.